esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -0,0 +1,865 @@
+            toXML(oldEnvironment.get(), defxml,0,0);
+            root->removeTree(oldEnvironment);
+            conn->commit();
+            ret = true;
+        }
+        else
+        {
+            errmsg.append("Could not find ESDL definition. Verify Id (name.version).");
+        }
+        conn->close();
+        return ret;
+    }
+
+    virtual bool deleteBinding(const char* bindingId, StringBuffer& errmsg, StringBuffer& bindingxml)
+    {
+        if (!bindingId || !*bindingId)
+            return false;
+        bool ret = false;
+        Owned<IRemoteConnection> conn = querySDS().connect(ESDL_BINDINGS_ROOT_PATH, myProcessSession(), 0, SDS_LOCK_TIMEOUT_DESDL);
+        if (!conn)
+            throw MakeStringException(-1, "Unable to connect to %s dali path", ESDL_BINDINGS_ROOT_PATH);
+
+        Owned<IPropertyTree> root = conn->getRoot();
+        if (!root)
+            throw MakeStringException(-1, "Unable to open %s dali path", ESDL_BINDINGS_ROOT_PATH);
+
+        VStringBuffer xpath("%s[@id='%s']", ESDL_BINDING_ENTRY, bindingId);
+        Owned<IPropertyTree> oldEnvironment = root->getPropTree(xpath.str());
+        if (oldEnvironment.get())
+        {
+            toXML(oldEnvironment.get(), bindingxml,0,0);
+            root->removeTree(oldEnvironment);
+            conn->commit();
+            ret = true;
+        }
+        else
+        {
+            errmsg.append("Could not find ESDL Binding.");
+        }
+        conn->close();
+        return ret;
+    }
+
+    virtual IPropertyTree* getDefinitions()
+    {
+        Owned<IRemoteConnection> conn = querySDS().connect(ESDL_DEFS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
+        if (!conn)
+           throw MakeStringException(-1, "Unable to connect to ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
+
+        conn->close(false); //release lock right away
+
+        Owned<IPropertyTree> esdlDefinitions = conn->getRoot();
+        if (!esdlDefinitions.get())
+           throw MakeStringException(-1, "Unable to open ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
+
+        return esdlDefinitions.getLink();
+    }
+
+    virtual IPropertyTree* getBindings()
+    {
+        Owned<IRemoteConnection> conn = querySDS().connect(ESDL_BINDINGS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
+        if (!conn)
+           throw MakeStringException(-1, "Unable to connect to ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
+
+        conn->close(false); //release lock right away
+
+        Owned<IPropertyTree> esdlBindings = conn->getRoot();
+        if (!esdlBindings.get())
+           throw MakeStringException(-1, "Unable to open ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
+        return esdlBindings.getLink();
+    }
+};
+
+class CEsdlSDSSubscription : implements IEsdlSubscription, public CInterface, implements ISDSSubscription
```

Comment:
The other improvement @jakesmith mentioned was to minimize/eliminate the dali subscriptions all together. Instead of relying on the dali notifications, wsesdlconfig could communicate the esdl def/binding change directly to the appropriate DESDL binding/service.

Reply:
@rpastrana again this issue is simply create the interface and set up the stage for other improvements. I've created another task for the subscription improvement: https://track.hpccsystems.com/browse/HPCC-18754, as a subtask of 18153.

Reply:
@mayx - it may be beyond the scope of this PR, but from a brief look at it, you're new interfaces are tied to a subscription / notify paradigm.
Is the intention to then remove these at later point, when the updates are pushed without a subscription?

Reply:
@jakesmith The new subscription interface is to isolate the components that will act on the changes(consumer), and the generic component that will receive the changes(producer). Currently the producer uses Dali subscription and notification, but it can also be a service that receives pushes. The Subscription is simply a bridge between the consumer and the producer for them to communicate asynchronously. Some function parameters (for example espprocess and binding) may change a bit later with other planned changes, but the paradigm should work with pushing notifications.